### ISSUE TITLE
Mismatch produced by Certificate Serial Numbers

### ISSUE THREAD

**Issue by patricksvgr** [fhightower/ioc-finder#96]
When parsing a big-endian hexadecimal representation of a [X.509 Certificate Serial Number](https://tools.ietf.org/html/rfc3280#section-4.1.2.2), e.g.

> "SolarWinds.Orion.Core.BusinessLayer.dll is signed by SolarWinds, using the certificate with serial number 0f:e9:73:75:20:22:a6:06:ad:f2:a3:6e:34:5d:c0:ed. The file was signed on March 24, 2020." ([Source](https://www.fireeye.com/blog/threat-research/2020/12/evasive-attacker-leverages-solarwinds-supply-chain-compromises-with-sunburst-backdoor.html))

the parser's `ioc_finder.find_iocs()` returns:
```
"ipv6s": [
    "ad:f2:a3:6e:34:5d:c0:ed",
    "0f:e9:73:75:20:22:a6:06"
    ],
"mac_addresses": [
    "0f:e9:73:75:20:22",
    "a6:06:ad:f2:a3:6e"
    ],
```
Some quickly collected ideas
- IOCs may be delimited by whitespaces (also, stripped by punctuation marks)
- a length limit for `ipv6s` and `mac_addresses` according to their format standards
- add a new IOCs type for _Certificate Serial Numbers_
- some "override/guess" strategy for multiple matching IOCs

Thanks and keep up the good work. 👍

**Comment by fhightower** [comment]
Thanks for reporting this @patricksvgr ! I'll look into this and identify what changes need to be made to fix this issue and better support certificate serial numbers.

**Comment by fhightower** [comment]
I hope to have a fix for this issue this Monday (Jan 11)

**Comment by fhightower** [comment]
@patricksvgr : Thanks for raising this issue! I pushed a new version of this library (`5.0.1`) with updated grammars for `ipv6s` and `mac_addresses` so that they will have better boundaries and will not parse anything they should not.

As to parsing certificate serial numbers, I haven't implemented it yet, but am tracking this with #98 (feel free to give it a 👍  if you would like to have it implemented).

Thanks again!

**Comment by patricksvgr** [comment]
Great, thank you! I'm going to check it within the next days.
